File: python/questions/maxLevelSum.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        mas = -inf
        tnl = [root]
        level = 1
        while len(tnl):
            tmp = tnl
            tnl = []
            ts = 0
            for n in tmp:
                ts += n.val
                if n.left:
                    tnl.append(n.left)
                if n.right:
                    tnl.append(n.right)
            if ts > mas:
                mas = ts
                ans = level
            level += 1
        return ans

File: cpp/questions/maxLevelSum.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        int mas = INT_MIN;
        std::vector<TreeNode*> tnl = { root };
        int level = 1;
        int ans;
        while (!tnl.empty()) {
            auto tmp = std::move(tnl);
            int ts = 0;
            for (auto& n : tmp) {
                ts += n->val;
                if (n->left) {
                    tnl.push_back(n->left);
                }
                if (n->right) {
                    tnl.push_back(n->right);
                }
            }
            if (ts > mas) {
                mas = ts;
                ans = level;
            }
            level++;
        }
        return ans;
    }
};
